daemon/main.c

```diff
@@ -532,6 +532,28 @@ main_terminate(
 //  Set process scheduling policy/priority and Lock resident memory.
 //
 
+static int move_to_root_cgroup(void) {
+  FILE *f;
+
+  f = fopen("/sys/fs/cgroup/cpu/tasks", "w");
+  if (f == NULL) {
+    log_message(LOG_DEBUG, "Can't open cgroups tasks file for writing\n");
+    return(-1);
+  }
+
+  if (fprintf(f, "%d\n", getpid()) <= 0) {
+		log_message(LOG_DEBUG, "Can't write pid into cgroups tasks file\n");
```

Comment:
It is usually good practice to print the error code/reason along with a failure message (usually `strerror(errno)`, or using `perror`, etc.
Looking through `xha` the convention seems to be to print the `errno` number as `(sys %d)`, please do that for each failure message.